app/auth/views.py

```diff
@@ -110,7 +110,10 @@ def request_reset_password():
         return redirect(url_for('main.dashboard'))
     form = ResetPasswordForm()
     if form.validate_on_submit():
-        user = User.query.filter_by(email = form.email.data).first_or_404()
+        user = User.query.filter_by(email = form.email.data).first()
+        if user is None:
+            flash('Invalid email. Please enter a valid email address', 'info')
+            return redirect(url_for('main.request_reset_password'))
```

Comment:
Since this is already done in the form validation, it can be removed.
```suggestion
```
See `auth/forms.py`
```
class ResetPasswordForm(FlaskForm, EmailFormMixin):
    submit = SubmitField('Reset Password')

    def validate_email(self, email):
        user = User.query.filter_by(email=email.data).first()
        if user is None:
            raise ValidationError('There is no account with the email you have entered. \
            Please ensure you have entered the correct email address or create a new account.')
```